### PR TITLE
Corinne - l131 nonco2 proc r s t y

### PR DESCRIPTION
**Pull request by cahartin**
#128

### CODE REVIEW

File: R/constants.R [hunk 1]
```diff
@@ -76,7 +76,7 @@ CONV_T_METRIC_SHORT <- 1000/908  # Ratio between metric ton and short ton
 CONV_MCAL_PCAL <- 1e-9
 CONV_HA_BM2 <- 1e-5
 CONV_THA_KGM2 <- 0.1   # tons C/ha -> kg C/m2
-CONV_GG_TG <- 0.001
+CONV_GG_TG <- 0.001 #gigagrams to tegagrams
```
Comment (bpbond) [inline R/constants.R]: Space after `#` please

File: R/zchunk_L131.nonco2_proc_R_S_T_Y.R [hunk 2]
```diff
@@ -1,26 +1,30 @@
 #' module_emissions_L131.nonco2_proc_R_S_T_Y
 #'
-#' Briefly describe what this chunk does.
+#' Calculate historical emissions from the processing sector by GCAM technology,computed from EDGAR emissions data and EPA emissions factors.
```
Comment (bpbond) [inline R/zchunk_L131.nonco2_proc_R_S_T_Y.R]: "technology, computed"

File: R/zchunk_L131.nonco2_proc_R_S_T_Y.R [hunk 3]
```diff
@@ -36,53 +40,155 @@ module_emissions_L131.nonco2_proc_R_S_T_Y_DISABLED <- function(command, ...) {
...
-      add_title("descriptive title of data") %>%
-      add_units("units") %>%
-      add_comments("comments describing how data generated") %>%
-      add_comments("can be multiple lines") %>%
+    EPA_Ind <- get_data(all_data, "emissions/EPA_FCCC_IndProc_2005")
+
+    # EDGAR emissions data
+    get_data(all_data, "emissions/EDGAR_NH3") %>%
+      gather(year, value, -`IPCC-Annex`, -`World Region`, -ISO_A3, -Name, -IPCC, -IPCC_description) %>%
+      mutate(Non.CO2='NH3') ->
```
Comment (bpbond) [inline R/zchunk_L131.nonco2_proc_R_S_T_Y.R]: In general please have spaces around mathematical operators like `+`, `=`, etc.

File: R/zchunk_L131.nonco2_proc_R_S_T_Y.R [hunk 4]
```diff
@@ -36,53 +40,155 @@ module_emissions_L131.nonco2_proc_R_S_T_Y_DISABLED <- function(command, ...) {
...
+
+    # First: Compute subsector share of sectoral emissions using EPA data
+
+    EPA_Ind %>%
+      left_join(EPA_tech, by = "Source_Category") %>%
+      rename(subsector = fuel) %>%
+      select(-EPA_Source_Category_Raw, -Source_Category) %>%   # Remove unnecessary columns
+      gather(Non.CO2, value, -sector,-subsector) %>%  # make a new column called Non.co2
+      na.omit() %>%  # delete rows with NAs
+      group_by(sector,subsector,Non.CO2) %>% # group by sector, subsector and Non.CO2
```
Comment (bpbond) [inline R/zchunk_L131.nonco2_proc_R_S_T_Y.R]: Spaces after commas too please

File: R/zchunk_L131.nonco2_proc_R_S_T_Y.R [hunk 5]
```diff
@@ -36,53 +40,155 @@ module_emissions_L131.nonco2_proc_R_S_T_Y_DISABLED <- function(command, ...) {
...
+      summarise(x=sum(value)) -> #sum the values
+      L131.EPA_nonco2_indproc.mlt
+
+    # Second: Compute share of sectoral emissions in each subsector using EPA data
+    GCAM_sector_tech %>%
+      select(supplysector, subsector, stub.technology, EPA_agg_sector, EPA_agg_fuel_ghg, EDGAR_agg_sector) %>%
+      filter(EDGAR_agg_sector %in% c("industry_processes", "chemicals", "landfills", "wastewater","aerosols",
```
Comment (bpbond) [inline R/zchunk_L131.nonco2_proc_R_S_T_Y.R]: What's special about these sectors? _Why_ are we filtering?

File: R/zchunk_L131.nonco2_proc_R_S_T_Y.R [hunk 6]
```diff
@@ -36,53 +40,155 @@ module_emissions_L131.nonco2_proc_R_S_T_Y_DISABLED <- function(command, ...) {
...
+      repeat_add_columns(tibble::tibble(Non.CO2=c("CH4","N2O","NMVOC","NOx","SO2","CO","VOC"))) ->
+      L131.nonco2_pct_R_prc_S_S_2005
+
+    # Third: Group by and then join by sector, subsector, and Non.CO2.  However, to join the columns they have to have
+    # the same name. Rename column to tech_emissions. Tech_emissions is the sum and tech_share is the tech_emissions/sector_emissions
+    L131.nonco2_pct_R_prc_S_S_2005 %>%
+      group_by(EPA_agg_sector,EPA_agg_fuel_ghg,Non.CO2) %>%
+      left_join(L131.EPA_nonco2_indproc.mlt, by=c(EPA_agg_sector="sector", EPA_agg_fuel_ghg = "subsector","Non.CO2")) %>%
+      rename(tech_emissions = x) %>%
+      mutate(tech_emissions=if_else(is.na(tech_emissions),1,tech_emissions)) ->
```
Comment (bpbond) [inline R/zchunk_L131.nonco2_proc_R_S_T_Y.R]: Why are we replacing NAs with 1s here?

File: R/zchunk_L131.nonco2_proc_R_S_T_Y.R [hunk 7]
```diff
@@ -36,53 +40,155 @@ module_emissions_L131.nonco2_proc_R_S_T_Y_DISABLED <- function(command, ...) {
...
+      left_join(L131.nonco2_gg_R_prc_S_2005, by =c("EPA_agg_sector", "Non.CO2")) %>%
+      rename(sector_emissions = tech_emissions.y ) %>% #sector emisisons from thr join
+      rename(tech_emissions = tech_emissions.x) %>%
+      mutate(tech_share = tech_emissions/sector_emissions) ->
+      L131.nonco2_pct_R_prc_S_S_2005
+
+    # Fourth: Disaggregate EDGAR emissions to subsectors
+    # First aggregate all EDGAR data and subset for processing sectors
+
+    #filter out years 2009 and 2010 in EDGAR_VOC
```
Comment (bpbond) [inline R/zchunk_L131.nonco2_proc_R_S_T_Y.R]: What's special about 2009-2010? Why are we doing this?

File: R/zchunk_L131.nonco2_proc_R_S_T_Y.R [hunk 8]
```diff
@@ -36,53 +40,155 @@ module_emissions_L131.nonco2_proc_R_S_T_Y_DISABLED <- function(command, ...) {
...
+
+
+    # combine all of the nonco2s together in one file
+    bind_rows(EDGAR_CH4, EDGAR_N2O, EDGAR_VOC, EDGAR_NOx, EDGAR_SO2, EDGAR_CO, EDGAR_NH3) %>%
+      left_join(EDGAR_sector, by=c("IPCC_description", "IPCC")) %>%
+      left_join(iso_GCAM_regID, by=c(ISO_A3 = "iso")) %>%
+      filter(agg_sector %in% c("industry_processes", "chemicals", "landfills", "wastewater","aerosols",
+                               "metals","foams","solvents", "semiconductors")) %>%
+      select(year,value, Non.CO2,agg_sector,GCAM_region_ID) %>%
+      filter(complete.cases(.)) %>%  #delete rows with NAs
```
Comment (bpbond) [inline R/zchunk_L131.nonco2_proc_R_S_T_Y.R]: Huh, I did not know you could do this.

File: R/zchunk_L131.nonco2_proc_R_S_T_Y.R [hunk 9]
```diff
@@ -36,53 +40,155 @@ module_emissions_L131.nonco2_proc_R_S_T_Y_DISABLED <- function(command, ...) {
...
+      group_by(GCAM_region_ID, year, Non.CO2, agg_sector) %>%
+      summarise(EDGAR_emissions=sum(value)) %>%
+      mutate(EDGAR_emissions=EDGAR_emissions * CONV_GG_TG) -> #convert from gg to tg
+      L131.EDGAR.mlt
+
+    L131.EDGAR.mlt$GCAM_region_ID <- as.integer(L131.EDGAR.mlt$GCAM_region_ID)
```
Comment (bpbond) [inline R/zchunk_L131.nonco2_proc_R_S_T_Y.R]: Might put these two lines as end of pipeline above, using `mutate`?

File: R/zchunk_L131.nonco2_proc_R_S_T_Y.R [hunk 10]
```diff
@@ -36,53 +40,155 @@ module_emissions_L131.nonco2_proc_R_S_T_Y_DISABLED <- function(command, ...) {
...
+      repeat_add_columns(tibble::tibble(year= c(emissions.EDGAR_YEARS))) %>%
+      repeat_add_columns(tibble::tibble(GCAM_region_ID = c(GCAM_region_names$GCAM_region_ID))) %>%
+      group_by(GCAM_region_ID, EDGAR_agg_sector.x, Non.CO2, year) %>%
+      left_join(L131.EDGAR.mlt, by = c("GCAM_region_ID", "EDGAR_agg_sector.x" ="agg_sector", "Non.CO2", "year")) %>%
+      na.omit() %>%  #delete rows with NAs
+      mutate(input.emissions = EDGAR_emissions * tech_share) %>%
+      group_by(GCAM_region_ID, supplysector, subsector, stub.technology, Non.CO2, year) %>%
+      summarise(input.emissions=sum(input.emissions)) %>%
+
+      # Produce outputs\
```
Comment (bpbond) [inline R/zchunk_L131.nonco2_proc_R_S_T_Y.R]: Remove extraneous backslash

File: R/zchunk_L131.nonco2_proc_R_S_T_Y.R [hunk 11]
```diff
@@ -36,53 +40,155 @@ module_emissions_L131.nonco2_proc_R_S_T_Y_DISABLED <- function(command, ...) {
...
+                     "emissions/GCAM_sector_tech",
+                     "emissions/EDGAR_NH3",
+                     "emissions/EDGAR_CH4",
+                     "emissions/EDGAR_N2O",
+                     "emissions/EDGAR_NMVOC",
+                     "emissions/EDGAR_NOx",
+                     "emissions/EDGAR_SO2",
+                     "emissions/EDGAR_CO",
+                     "emissions/EPA_FCCC_IndProc_2005") %>%
       # typical flags, but there are others--see `constants.R`
```
Comment (bpbond) [inline R/zchunk_L131.nonco2_proc_R_S_T_Y.R]: Remove

File: inst/extdata/emissions/EDGAR_sector.csv [hunk 12]
```diff
@@ -0,0 +1,53 @@
+# File: EDGAR_sector.csv,,
+# Title: Mapping file from IPCC to agg sector,,
+# Units: NA,,
+# Comments: Comments, may be multiline,,
+# Source: Citation or source if available,,
```
Comment (bpbond) [inline inst/extdata/emissions/EDGAR_sector.csv]: Remove these two lines

File: R/zchunk_L131.nonco2_proc_R_S_T_Y.R [hunk 13]
```diff
@@ -36,53 +40,155 @@ module_emissions_L131.nonco2_proc_R_S_T_Y_DISABLED <- function(command, ...) {
...
       add_flags(FLAG_NO_TEST, FLAG_LONG_YEAR_FORM, FLAG_NO_XYEAR) ->
```
Comment (bpbond) [inline R/zchunk_L131.nonco2_proc_R_S_T_Y.R]: You need to remove the `FLAG_NO_TEST`! If this is still here, it means the chunk's output wasn't getting run through the testing system.

Comment (cahartin) [inline R/zchunk_L131.nonco2_proc_R_S_T_Y.R]: oh no! I'm working off of an older wiki print out. grrr!

**Comment by bpbond** [comment]
Hey @cahartin - I'm going to close this for now. Re-open when you're ready. Thanks!

**Pull request closed without merging**